tests/test_caching.py

```diff
@@ -492,8 +492,8 @@ def test_sparse_function(self, operate_on_empty_cache):
 
         u = SparseFunction(name='u', grid=grid, npoint=1, nt=10)
 
-        # created: u, p_u, h_p_u, u_coords, d, h_d
-        ncreated = 6
+        # created: u, u(inds), p_u, h_p_u, u_coords, u_coords(inds), d, h_d
+        ncreated = 8
```

Comment:
since we now cache both `u` and `u(x)` to avoid recreating objects, this creates to cache entry per Function the first time. Same below and other tests